packages/web-components/fast-foundation/CEMToMarkdown.mjs

```diff
@@ -0,0 +1,176 @@
+            componentManifest.modules.push(modules[i - 1]);
+            componentManifest.modules.push(modules[i - 2]);
+        }
+
+        // enclose html tags in `` to prevent tags in comments from confusing docusaurus
+        // and remove new line characters from descriptions
+        componentManifest.modules.forEach(module=>{
+            module.declarations?.forEach(dec=>{
+                if(dec.description)
+                {
+                    dec.description = fixTagsInText(dec.description.replaceAll(LF, ' '));
+                }
+                if(dec.default)
+                {
+                    dec.default = fixTagsInText(dec.default.replaceAll(LF, ' '));
+                }
+                dec.members?.forEach(member=>{
+                    if(member.description)
+                    {
+                        member.description = fixTagsInText(member.description.replaceAll(LF, ' '));
+                    }
+                    if(member.default)
+                    {
+                        member.default = fixTagsInText(member.default.replaceAll(LF, ' '));
+                    }
+                    if(member.return?.type?.text)
+                    {
+                        // these are already rendered inside of back-ticks so we only need to remove new lines
+                        member.return.type.text = member.return.type.text.replaceAll(LF, ' ');
+                    }
+                });
+                dec.attributes?.forEach(attr=>{
+                    if(attr.description)
+                    {
+                        attr.description = fixTagsInText(attr.description.replaceAll(LF, ' '));
+                    }
+                });
+            })
+        });
+
+        // Convert the single component manifest into a markdown string.
+        let markdown = customElementsManifestToMarkdown(componentManifest,
+            {
+                headingOffset: 1,
+                private: 'hidden'
+            });
+
+        // Replace our < and > markers with backticks and < >
+        // This is necessary because customElementsManifestToMarkdown escapes the backticks during the conversion
+        // and we don't want that because then docusaurus will see the tags as real tags instead of just text.
+        markdown = markdown.replaceAll("REPLACELT","`<").replaceAll("REPLACEGT",">`");
+
+        // Get the README.md file
+        let path = modules[componentIndex].path.split('/');
+        path[path.length - 1] = "README.md";
+        path = path.join('/');
+
+        // If a README.md file exists
+        if(fs.existsSync(path))
+        {
+            // Read the contents of the file
+            let readMe = fs.readFileSync(path, 'utf-8');
+
+            // Find the location of the "## API" section
+            let apiLoc = readMe.indexOf("## API");
+
+            // Find the location of the "## Additional resources" section
+            let resourcesLoc = readMe.indexOf("## Additional resources");
```

Comment:
I think you should be able to find "the next h2" as in the next heading with that double hash and use that instead in case new sections arise. That might make this a little more robust.

Reply:
This is trickier than it seems as the generated markdown contains "h2" headings. I looked into using HTML comments as markers but it seems that they are not universally supported in markdown. So the rule is that the API section will be placed before the "Additional resources" section if included otherwise at the end of the file.